core/amber/src/main/scala/edu/uci/ics/texera/web/service/ExecutionResultService.scala

```diff
@@ -250,7 +250,13 @@ class ExecutionResultService(
                 oldInfo.tupleCount,
                 info.tupleCount
               )
-              if (StorageConfig.resultStorageMode == ICEBERG) {
+
+              val outputPortsMap = physicalPlan.getPhysicalOpsOfLogicalOp(opId).headOption.map(_.outputPorts).getOrElse(Map.empty)
+              val hasSingleSnapshot = outputPortsMap.values.exists { case (outputPort, _, _) =>
+                outputPort.mode == OutputMode.SINGLE_SNAPSHOT
+              }
+
+              if (StorageConfig.resultStorageMode == ICEBERG && !hasSingleSnapshot) {
                 // using the first port for now. TODO: support multiple ports
```

Comment:
move this comment to line 253. Also, add a comment to line 256 to say SINGLE_SNAPSHOT is used for HTML content.